server/src/services/organizations.service.ts

```diff
@@ -539,4 +551,101 @@ export class OrganizationsService {
     }
     return organizationUser;
   }
+
+  decamelizeDefaultGroupNames(groups) {
+    return groups
+      .split('|')
+      .map((group: string) =>
+        group === 'All Users' || group === 'Admin' ? decamelize(group.replace(' ', '')) : group
+      );
+  }
+
+  async bulkUploadUsers(currentUser: User, fileStream, res: Response) {
+    const users = [];
+    const existingUsers = [];
+    const invalidRows = [];
+    const invalidGroups = [];
+    const emailPattern = /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i;
+
+    csv
+      .parseString(fileStream.toString(), {
+        headers: ['first_name', 'last_name', 'email', 'groups'],
+        renameHeaders: true,
+        ignoreEmpty: true,
+      })
+      .transform((row: UserCsvRow, next) => {
+        return next(null, {
+          ...row,
+          groups: this.decamelizeDefaultGroupNames(row?.groups),
+        });
+      })
+      .validate(async (data: UserCsvRow, next) => {
+        await dbTransactionWrap(async (manager: EntityManager) => {
+          //Check for existing users
+          const user = await this.usersService.findByEmail(data?.email);
```

Comment:
Shall we use same `manager` for all queries?

Reply:
Yup I will add the manager from  getManager() here

Reply:
You can use the manager 
```
await dbTransactionWrap(async (manager: EntityManager) => {
```